Added warnings to the provider registry docs (#405)
spacelift-io/user-documentation: docs/vendors/terraform/provider-registry.md

}
```

!!! warning

>     The type attribute in the `spacelift_terraform_provider` resource is actually referring to the unique name for the provider within Spacelift account. It must consist of lowercase letters only.

Reviewer: ```suggestion
    The `type` attribute in the `spacelift_terraform_provider` resource refers to the unique name of the provider within Spacelift account. It must consist of lowercase letters only.
```

Author: Done

spacectl provider list-versions --type=$YOUR_PROVIDER_TYPE
```

!!! warning

>     The type parameter in the `spacectl provider list-versions` command is actually referring to the unique name for the provider within Spacelift account. It must consist of lowercase letters only.

Reviewer: ```suggestion
    The `type` parameter in the `spacectl provider list-versions` command refers to the unique name of the provider within Spacelift account. It must consist of lowercase letters only.
```

Author: Done

Note the version status. Versions start their life as drafts, and you can publish them by grabbing their ID (first column) and using `spacectl`:

```bash
spacectl provider publish-version --version=$YOUR_VERSION_ID
```

!!! warning

>     The version parameter in the `spacectl provider publish-versions` command is actually referring to the unique ID for the provider version within Spacelift account.

Reviewer: ```suggestion
    The `version` parameter in the `spacectl provider publish-versions` command refers to the unique ID of the provider version within Spacelift account.
```

Author: Done
